Cypress/E2E: Promote and Update

furqanmlk opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->
Promoted ```custom_emoji_2_spec.js``` to prod
Added unstable test case to ```custom_emoji_3_spec.js``` 
Fixed ```MM-T1391 Verify the manage options displayed for Guest User```

josephbaylon reviewed (changes requested):
@furqanmlk Please fix lint issues

josephbaylon reviewed (approved):
LGTM

prapti reviewed (changes requested):
Hi Furqan,
I'm not too sure about promoting `custom_emoji_3` yet. In the last 5 runs, looks like the only test that's been stable is T2183.
T2184 has failed in 2 runs (builds 4061 and 3972).
T2185 has failed in 2 runs (builds 4012 and 3948).
Should we wait on promoting that spec until we take a deeper look?

furqanmlk commented:
Hi @prapti, 
I think you misunderstood. 
custom_emoji_3_spec.js did get promoted. I added unstable test cases to this spec.

> Hi Furqan,
> 
> I'm not too sure about promoting `custom_emoji_3` yet. In the last 5 runs, looks like the only test that's been stable is T2183.
> 
> T2184 has failed in 2 runs (builds 4061 and 3972).
> 
> T2185 has failed in 2 runs (builds 4012 and 3948).
> 
> Should we wait on promoting that spec until we take a deeper look?

prapti commented:
Oh sorry @furqanmlk I meant `custom_emoji_2` (not `custom_emoji_3`). Since you're promoting that spec, and only one test in that file (T2183) seems stable but not the other two (T2184 which is still in that file, and T2185 which you're moving the unstable `custom_emoji_3`) are both quite unstable.

saturninoabril reviewed (approved):
Not sure if moving around test case (like `MM-T2185`) is something that we should normally do. In my mind, move test either a passing/failing test into a new spec file to isolate but a fix should be made to the failing test on the same PR, unless such failed test is a known issue like product bug, or any limitation in our test infra.

I'm good as is but open for discussion as a group.

furqanmlk commented on e2e/cypress/integration/emoji/custom_emoji_2_spec.js:
```diff
@@ -7,6 +7,7 @@
 // - Use element ID when selecting an element. Create one if none.
 // ***************************************************************
 
+
```
```suggestion
```

furqanmlk commented:
> Oh sorry @furqanmlk I meant `custom_emoji_2` (not `custom_emoji_3`). Since you're promoting that spec, and only one test in that file (T2183) seems stable but not the other two (T2184 which is still in that file, and T2185 which you're moving the unstable `custom_emoji_3`) are both quite unstable.

Thanks @prapti 
Yes you are right, I moved T2183 to wrong file. 
I have moved T2183 to `custom_emoji_1` as it is now stable, rest I did not change.

furqanmlk commented:
> Not sure if moving around test case (like `MM-T2185`) is something that we should normally do. In my mind, move test either a passing/failing test into a new spec file to isolate but a fix should be made to the failing test on the same PR, unless such failed test is a known issue like product bug, test compatibility issue among server edition/license (Cloud, E20, TE) or a limitation in our test infra.
> 
> I'm good as is but open for discussion as a group.

Thanks @saturninoabril , yes you are right I misplaced the stable test case to wrong file.
everything is back to original file except T2183, it has moved to `custom_emoji_1` as it is stable.

prapti reviewed (approved):
Agree with Saturn's argument.
Looks good for now.
Thank you for fixing it @furqanmlk!
